test/services/AccountDetailsServiceSpec.scala

```diff
@@ -32,28 +32,24 @@ import scala.concurrent.ExecutionContext.Implicits.global
 
 class AccountDetailsServiceSpec extends AnyWordSpecLike with MockFactory with Matchers {
 
-  private trait Test {
-    val customerInfoResult: HttpGetResult[CustomerInformation] = Right(customerInformationMax)
+  lazy val customerInfoResult: HttpGetResult[CustomerInformation] = Right(customerInformationMax)
 
-    implicit val hc: HeaderCarrier = HeaderCarrier()
-    val mockVatSubscriptionConnector: VatSubscriptionConnector = mock[VatSubscriptionConnector]
+  implicit val hc: HeaderCarrier = HeaderCarrier()
+  val mockVatSubscriptionConnector: VatSubscriptionConnector = mock[VatSubscriptionConnector]
 
-    def setup(): Any = {
-      (mockVatSubscriptionConnector.getCustomerInfo(_: String)(_: HeaderCarrier, _: ExecutionContext))
-        .expects(*, *, *)
-        .returns(Future.successful(customerInfoResult))
-    }
-
-    def target: AccountDetailsService = {
-      setup()
-      new AccountDetailsService(mockVatSubscriptionConnector)
-    }
+  def accountDetailsService: AccountDetailsService = {
+    new AccountDetailsService(mockVatSubscriptionConnector)
   }
 
   "Calling .getAccountDetails" should {
 
-    "retrieve the customer details" in new Test {
-      val details: HttpGetResult[CustomerInformation] = await(target.getAccountDetails("123456789"))
+    "retrieve the customer details" in {
+      val details: HttpGetResult[CustomerInformation] = {
+        (mockVatSubscriptionConnector.getCustomerInfo(_: String)(_: HeaderCarrier, _: ExecutionContext))
```

Comment:
Could you put this mocking in a little function in this file? Just conscious it's being repeated in every test here with the only change being the value that is returned (I don't think the VRN matters).